sw/device/tests/crypto/cryptotest/firmware/extclk_sca_fi.c

```diff
@@ -0,0 +1,76 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/tests/crypto/cryptotest/firmware/extclk_sca_fi.h"
+
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_clkmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/ottf_test_config.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/ujson/ujson.h"
+#include "sw/device/tests/crypto/cryptotest/firmware/status.h"
+#include "sw/device/tests/crypto/cryptotest/json/extclk_sca_fi_commands.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_clkmgr_t clkmgr;
+
+/**
+ * Extclk Configure command handler.
+ *
+ * Allows to configure the external clock.
+ * The uJSON data contains:
+ *  - sel: External clock on or off.
+ *  - hi_speed_sel: Nominal or low-speed external clock.
+ *
+ * @param uj The received uJSON data.
+ */
+status_t handle_extclk_sca_fi_configure(ujson_t *uj) {
+  cryptotest_extclk_sca_fi_cfg_t uj_data;
+  TRY(ujson_deserialize_cryptotest_extclk_sca_fi_cfg_t(uj, &uj_data));
+
+  UJSON_CHECK_DIF_OK(dif_clkmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR), &clkmgr));
+
+  bool is_low_speed = true;
+  if (uj_data.hi_speed_sel == 1) {
+    is_low_speed = false;
+  }
+
+  if (uj_data.sel == 0) {
+    // Disable external clock.
+    UJSON_CHECK_DIF_OK(dif_clkmgr_external_clock_set_disabled(&clkmgr));
+  } else {
+    // Enable external clock.
+    UJSON_CHECK_DIF_OK(
+        dif_clkmgr_external_clock_set_enabled(&clkmgr, is_low_speed));
+  }
+  // Check CLKMGR_EXTCLK_STATUS_REG_OFFSET register. Blocks if clock does not
+  // switch.
+  UJSON_CHECK_DIF_OK(dif_clkmgr_wait_for_ext_clk_switch(&clkmgr));
+
+  return OK_STATUS(0);
```

Comment:
nit: you can just use `return OK_STATUS();` the added zero does not add much: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/base/status.h;drc=e604dadf38028db25218bb373da1ad14914a85f7;l=258

(applies throughout)

Reply:
Done, thanks.